blocks/bricks/cost.py

```diff
@@ -18,6 +18,19 @@ def apply(self, y, y_hat):
 
 
 @add_metaclass(ABCMeta)
+class Softmax(Brick):
+    @application
+    def apply(self, x):
+        return tensor.nnet.softmax(x)
+
+    @application
+    def categorical_cross_entropy(self, x, y_hat):
```

Comment:
Also, could you add a docstring that describes what the inputs should be e.g. `y` can be of the same dimension (in which case it needs to be a full distribution) or it needs to have one less dimension (in which case it needs to have integers representing the target classes).